### PR TITLE
[inductor][non determinism] Disable autotuning when determinisitic mode is ON

### PR DESCRIPTION
**Pull request by anijain2305**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #99851
* #99793

This removes a source of non-determinism seen in `sebotnet33ts_256`. However, we should see if we can reduce non-determinism in autotuning in general.

cc @soumith @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

**Comment by anijain2305** [comment]
Inductor run here - https://github.com/pytorch/pytorch/actions/runs/4781747485

**Review by ngimel — APPROVED** [review]

**Comment by desertfire** [comment]
Should we introduce a flag to pick a deterministic triton-generated code (e.g. based on a sorting of hashcode) and enable that for accuracy testing?

**Comment by ngimel** [comment]
We can just disable autotuning (there's a flag for that), that should be deterministic

**Review by jansel — CHANGES_REQUESTED** [review]

File: torch/_inductor/triton_heuristics.py [hunk 1]
```diff
@@ -810,11 +819,13 @@ def persistent_reduction(size_hints, reduction_hint=False, meta=None, filename=N
                 size_hints, 2 * (256 // rnumel) if rnumel <= 256 else 1, rnumel
             )
         ]
-
     for c in configs:
         # we don't need RBLOCK for persistent reduction
         c.kwargs.pop("RBLOCK")
 
+    if not config.triton.autotune_pointwise:
+        configs = configs[:1]
```
[the remark this answers is not in the document]
Comment (anijain2305) [reply torch/_inductor/triton_heuristics.py:827]: Oh, this returns only 1 config if autotune_pointwise if False. Basically disables autotuning.

File: torch/_inductor/triton_heuristics.py [hunk 2]
```diff
@@ -168,6 +168,15 @@ def launcher({', '.join(def_args)}, grid, stream):
...
         return launcher
 
+    def round_benched_timing(self, x, decimal_places=1):
+        """
+        Sometimes different autotuning configs can have very small runtime
+        difference. This can affect the ranking of different configs, making
+        autotuning nondeterministic.  This function rounds the benched timing to
+        a default value of one decimal place, i.e., precision of 0.1 ms.
+        """
+        return builtins.round(x, decimal_places)
```
Comment (jansel) [inline torch/_inductor/triton_heuristics.py:178]: Rounding could still be non-determinstic (just less likely).  Perhaps we should change the cache format to just store which one was selected?

**Review by jansel — APPROVED** [review]

**Review by desertfire — APPROVED** [review]
The new version LGTM.

**Pull request closed without merging**